### ISSUE TITLE
mocks and infinite nested $ref

### ISSUE THREAD

**Issue by cgarnier** [apigee-127/swagger-tools#396]
Mocks may generate bad values when $ref is pointing on its model.

```
{
  "type": "object",
  "description": "A pet",
  "properties": {
    "name": {
       "type": "text"
   },
    "children": {
      "type": "array",
      "items": {
         "$ref": "./shemas/pet.json"
      }
    }
  },
  "required": ["name"]
}
```
When nesting goes too deep, the ref is no more replaced by the schema but by an empty object. Probably to avoid an infinite loop.
That s fail because it s no more respect the schema. 
For the exemple above, it s not possible to replace a pet by an empty object cause his name is required.

I think it s better to remove the whole object then put an empty object.

**Comment by sherpya** [comment]
I've noticed that making cycles in refs (like you have in a database model) causes empty schema object, so the validator is not able to validate anymore

**Comment by cgarnier** [comment]
Yes but i think it s replacing object by empty object to avoid the infinite loop. I should be better to remove the object. This way it doesnt break the validation

**Comment by sherpya** [comment]
I've made a fitting that converts date and date-times to moment, but if a definition has a loop inside I don't get a schema in `req.swagger.params`

**Comment by whitlockjc** [comment]
I'm in the process of rewriting the json-refs resolution support, which is what swagger-tools/sway use.  json-refs' currently isn't handling circulars properly, at least not in all cases, and the case you bring up is the one that we're seeing.  This originally was reported here: https://github.com/apigee-127/sway/issues/85
